docs/Backup & restore.md

```diff
@@ -87,22 +75,34 @@ Devices MUST allow restoration of modified backups (full or partial backups wher
 
 | ![Restoring on a different device revision](images/restoring-same-dev-different-dev-rev.png) |
 |:--:|
 | _**Restoring on a different device revision**_ |
 
-Devices are RECOMMENDED to allow the restoration of backups created on a different device revision than their own.
-
 Devices MUST allow the partial restoration of backups which have at least one role path `status` of `Ok` when supplying the `allowPartial` argument of `true` in the request.
 
-## 4. Restoring a different instance of the same device type using the same device revision
+// TBD: Facilities can have a multitude of instances of the same device type. Backups performed on a device instance can be used to bootstrap other instances of the same device type. What if the new device overwrites things like IP addresses? Are we saying the restore should be used as a bootstrap mechanism (do we need a isTemplate boolean flag when restoring)?
 
-Facilities can have a multitude of instances of the same device type. Backups performed on a device instance can be used to bootstrap other instances of the same device type.
+## Definitions
+
+A `device`, for the purposes of this section, is a physical or logical entity that can be backed up and restored using the procedures described. It may or may not correspond to an IS-04 Device or IS-04 Node.
+
+`Backup data set` is the set of data retrieved from a device using the backup procedures described.
+
+`Device revision` represents any combination of software versions (this includes firmware) and hardware revision that dictates the functionality of the device. As a vendor evolves a product through its lifecycle the hardware, software and/or firmware that make up a device is likely to change. From time to time a vendor my issue new software or firmware that a customer can install that changes how a device behaves. A device may also have hardware such a peripherals or plug-in cards added or removed that change its behaviour. Each of these changes can affect how easily a `backup data set` can be restored to a device. The `device revision` is used to represent the ensemble of the versions of all the various components that affect a device’s functionality. It is up to a vendor to decide what changes to a device represent a change of `device revision`. A software upgrade will likely result in a device being considered to have a different `device revision` but a vendor may decide that a change in, for example, PCB colour does not represent a different `device revision`.
+
+`Backup validation fingerprint` is an optional string in a `backup data set` that can be used to capture the various versions of the hardware, software and/or firmware that made up a device at the time the backup was performed. The `backup validation fingerprint` can be used by a device to help decide whether the `backup data set` being restored to it is compatible. The format of the string is defined by the vendor and is opaque to other systems. This may contain information like:
```

Comment:
Remove normative language.
```suggestion
`Backup validation fingerprint` is an optional string in a `backup data set` that can be used to capture the various versions of the hardware, software and/or firmware that made up a device at the time the backup was performed. The `backup validation fingerprint` can be used by a device to help decide whether the `backup data set` being restored to it is compatible. The format of the string is defined by the vendor and is opaque to other systems. This could contain information like:
```